care/facility/migrations/0398_alter_patientconsultation_discharge_reason.py

```diff
@@ -0,0 +1,51 @@
+# Generated by Django 4.2.6 on 2023-12-06 09:17
+
+from django.db import migrations, models
+
+
+def migrate_discharge_reasons(apps, schema_editor):
+    PatientConsultation = apps.get_model("facility", "PatientConsultation")
+    mapping = {
+        "REC": 1,
+        "REF": 2,
+        "EXP": 3,
+        "LAMA": 4,
+    }
+
+    consultations_to_update = []
+    for consultation in PatientConsultation.objects.filter(
+        discharge_reason__isnull=False
+    ):
+        consultation.discharge_reason = mapping.get(consultation.discharge_reason, None)
+        consultations_to_update.append(consultation)
+
+    PatientConsultation.objects.bulk_update(
+        consultations_to_update, ["discharge_reason"]
+    )
```

Comment:
```suggestion
    mapping = {
        "REC": 1,
        "REF": 2,
        "EXP": 3,
        "LAMA": 4,
        "": None,
    }
    cases = [When(discharge_reason=k, then=Value(v)) for k, v in mapping.items()]

    PatientConsultation.objects.filter(
        discharge_reason__isnull=False
    ).update(
        discharge_reason=Case(*cases, default=Value(99))
    )
```
A single query solution
Instead of assigning random values to null, would it not be more appropriate to assign an unknown state?